erpnext_quota/erpnext_quota/quota.py

```diff
@@ -187,3 +188,32 @@ def get_directory_size(path):
             total_size += chr(char)
 
     return int(total_size)
+
+def document_limit(doc, event):
+    # store limits docs in dictionary
+    limit_dict = {}
+    for item in frappe.get_site_config()['quota']['document_limit']:
+        limit_dict[item['document_type']] = item
+
+    # match the limit based on period and limit number
+    if (limit_dict.get(doc.doctype)):
+        limit = frappe._dict(limit_dict.get(doc.doctype))
+        limit_period = get_limit_period(limit.period)
+        if len(frappe.db.get_list(doc.doctype, filters={
+            'creation': ['BETWEEN', [str(limit_period.start)+' 00:00:00.000000', 
+                str(limit_period.end)+' 23:59:59.999999']]
+            })) >= limit.limit:
+            msg = f"Limit exceeded for {doc.doctype}, {limit.period} limit is {limit.limit}. Please contact administrator."
+            frappe.throw(msg)
+
+def get_limit_period(period):
+    """
+        Get date mappinf for document limit period
+    """
+    today = str(getdate())
+    periods = {
+        'Daily': {'start': today, 'end': today},
+        'Weekly': {'start': add_days(today, -6), 'end': today},
```

Comment:
My suggestion is to keep it for the current week

Get first day of the week from frappe
`from frappe.utils import get_first_day_of_the_week`

Helping code
```
from datetime import date, timedelta

today = date.today()
start = today - timedelta(days=today.weekday())
end = start + timedelta(days=6)
```

Reply:
Done